src/plugins/data/server/search/opensearch_search/ext_opensearch_search_strategy.ts

```diff
@@ -0,0 +1,117 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import { first } from 'rxjs/operators';
+import { SharedGlobalConfig, Logger, SavedObject } from 'opensearch-dashboards/server';
+import { SearchResponse } from 'elasticsearch';
+import { Observable } from 'rxjs';
+import { ApiResponse, Client } from '@elastic/elasticsearch';
+import { SearchUsage } from '../collectors/usage';
+import { toSnakeCase } from './to_snake_case';
+import {
+  ISearchStrategy,
+  getDefaultSearchParams,
+  getTotalLoaded,
+  getShardTimeout,
+  shimAbortSignal,
+} from '..';
+
+export const extOpensearchSearchStrategyProvider = (
+  config$: Observable<SharedGlobalConfig>,
+  logger: Logger,
+  usage?: SearchUsage
+): ISearchStrategy => {
+  return {
+    search: async (context, request, options) => {
+      logger.debug(`search ${request.params?.index}`);
+      const config = await config$.pipe(first()).toPromise();
+      const uiSettingsClient = await context.core.uiSettings.client;
+
+      // Only default index pattern type is supported here.
+      // See data_enhanced for other type support.
+      if (!!request.indexType) {
+        throw new Error(`Unsupported index pattern type ${request.indexType}`);
+      }
+
+      // ignoreThrottled is not supported in OSS
+      const { ignoreThrottled, ...defaultParams } = await getDefaultSearchParams(uiSettingsClient);
+
+      const params = toSnakeCase({
+        ...defaultParams,
+        ...getShardTimeout(config),
+        ...request.params,
+      });
+
+      try {
+        // logger.info(JSON.stringify(params));
+        // logger.info(JSON.stringify(request.dataSource));
+        const dataSource = request.dataSource ? await context.core.savedObjects.client.get('data-source', request.dataSource) : undefined;
+        logger.info(JSON.stringify(dataSource));
+        const dataSourceObj = dataSource!.attributes as any;
+        const url = dataSourceObj.endpoint.url;
```

Comment:
This throws an exception when `dataSource` is undefined, right? Can we define the type for `dataSourceObj`?